src/replication.c

```diff
@@ -2134,37 +2606,68 @@ update_replica_status(spec_t *spec, zvol_io_hdr_t *hdr, replica_t *replica)
 
 			assert(r1 != NULL);
 			assert((spec->rebuild_info.dw_replica == replica) || (spec->replication_factor == 1));
-			spec->rebuild_info.dw_replica = NULL;
-			spec->rebuild_info.healthy_replica = NULL;
-			spec->rebuild_info.rebuild_in_progress = false;
 			if (found_in_list == 2) {
 				replica_count = spec->healthy_rcount + spec->degraded_rcount;
-				/* Error our if already quorum replicas are connected */
-				if (replica_count >= spec->replication_factor)
+
+				/* Error out if already quorum replicas are connected */
+				if (replica_count >= spec->desired_replication_factor) {
 					goto disconnect_all_non_quorum_replicas;
+				}
 			}
 			if (found_in_list == 1)
 				spec->degraded_rcount--;
 			else {
+				/* Point to transition replica */
+				spec->transition_replica = replica;
+				/* Pause ongoing IO's */
+				spec->quiesce = 1;
+				rc = wait_for_ongoing_ios_on_replica(spec, replica, pause_io_wait_time);
+				if (rc < 0) {
+					/* Resume ongoing IO's in case of error*/
+					spec->quiesce = 0;
+					ISTGT_ERRLOG("failed to flush the IO's on replica{%lu}"
+					    " (%s:%d) during replica mgmt_fd: %d set replica state as"
+					    " as %s\n", replica->zvol_guid, replica->ip,
+					    replica->port, replica->mgmt_fd, (replica->state
+					    == ZVOL_STATUS_HEALTHY) ? "healthy" : "degraded");
+					goto trigger_rebuild;
+				}
+				/* Update trusty replica list, replication and consistency factor*/
+				rc = update_replication_factor(spec, replica);
```

Comment:
increase of RF won't be needed everytime like for replica replacement, (or) for replica movement, RF won't be increased

Reply:
I need to update function name

Reply:
how about updating `update_replication_factor` as `update_unknown_as_trusty_replica` (which will update in etcd and in-memory structures based on whether it is scaleup case or replica movement/replacement case?